__tests__/app.test.js

```diff
@@ -443,6 +402,224 @@ describe("GET /api/users", () => {
+    return request(app)
+      .get("/api/articles?sort_by=article_img_url")
+      .expect(200)
+      .then(({ body }) => {
+        const { articles }= body;
+        expect(articles).toBeSortedBy("article_img_url", { descending: true } );
+      });
+  });
+  test("GET 200: responds with all articles ordered by comment_count as default order descending", () => {
+    return request(app)
+      .get("/api/articles?sort_by=comment_count")
+      .expect(200)
+      .then(({ body }) => {
+        const { articles }= body;
+        expect(articles).toBeSortedBy("comment_count", { descending: true } );
+      });
+  });
+  test("GET 200: responds with all articles ordered ascending", () => {
+    return request(app)
+      .get("/api/articles?order=asc")
+      .expect(200)
+      .then(({ body }) => {
+        const { articles }= body;
+        expect(articles).toBeSortedBy('created_at', { ascending: true } );
+      });
+  });
+  test("GET 200: responds with all articles ordered descending", () => {
+    return request(app)
+      .get("/api/articles?order=desc")
+      .expect(200)
+      .then(({ body }) => {
+        const { articles }= body;
+        expect(articles).toBeSortedBy('created_at', { descending: true } );
+      });
+  });
+  test("GET 200: responds with all queries", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch&sort_by=title&order=asc")
+      .expect(200)
+      .then(({ body }) => {
+        const { articles } = body;
+        expect(articles).toBeSortedBy('title', { ascending: true } );
+        expect(articles.length).toBe(11);
+        articles.forEach((article) => {
+          expect(article).toMatchObject({
+            author: expect.any(String),
+            title: expect.any(String),
+            article_id: expect.any(Number),
+            topic: 'mitch',
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            article_img_url: expect.any(String),
+            comment_count: expect.any(Number),
+          });
+        })
+      });
+  });
+  test("GET 404: responds with message and error if category to sort_by does not exist", () => {
+    return request(app)
+      .get("/api/articles?sort_by=carrots")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("status 404: not found");
+      });
+  });
+  test("GET 404: responds with error message if queried topic does not exist", () => {
+    return request(app)
+    .get("/api/articles?topic=dogs")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("status 404: not found");
+      });
+  });
+  test("GET 404: responds with message and error if there is a typo in the path", () => {
+    return request(app)
+      .delete("/api/ares?topic=dogs")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("status 404: Path not found");
+      });
+});
+})
+
```

Comment:
Could we just add the following two tests please?

- Status 400. invalid order query, e.g. ?order=bananas
- Status 200, valid topic query, but has no articles responds with an empty array of articles, e.g. ?topic=paper.